### PR TITLE
Add support for org.assertj.core.error.MultipleAssertionsError in test assertion failure events

### PR DESCRIPTION
**Pull request by donat**

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test qfl

**Comment by donat** [comment]
@bot-gradle test qfl

**Comment by donat** [comment]
@bot-gradle test rfm

File: subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r76/TestFailureProgressEventCrossVersionTest.groovy [hunk 1]
```diff
@@ -276,6 +278,8 @@ class TestFailureProgressEventCrossVersionTest extends ToolingApiSpecification {
             dependencies {
                 testImplementation 'org.junit.jupiter:junit-jupiter-api:5.7.1'
                 testImplementation 'org.junit.jupiter:junit-jupiter-engine:5.7.1'
+                testImplementation 'junit:junit:4.13.2'
```
Comment (marcphilipp) [inline subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r76/TestFailureProgressEventCrossVersionTest.groovy:281]: ❔ Is this actually required?

File: subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r76/TestFailureProgressEventCrossVersionTest.groovy [hunk 2]
```diff
@@ -299,6 +303,40 @@ class TestFailureProgressEventCrossVersionTest extends ToolingApiSpecification {
             }
         '''
 
+        file('src/test/java/org/gradle/AssertjMultipleFailureException.java') << '''
+            package org.gradle;
+
+            import org.assertj.core.error.MultipleAssertionsError;
+            import org.junit.jupiter.api.Test;
+            import org.junit.runners.model.MultipleFailureException;
```
Comment (marcphilipp) [inline subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r76/TestFailureProgressEventCrossVersionTest.groovy:311]: ❓ Unused?

**Review by marcphilipp — APPROVED** [review]

File: subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r76/TestFailureProgressEventCrossVersionTest.groovy (hunk 2 above)
Comment (donat) [reply subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r76/TestFailureProgressEventCrossVersionTest.groovy]: True, removed.

File: subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r76/TestFailureProgressEventCrossVersionTest.groovy (hunk 1 above)
Comment (donat) [reply subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r76/TestFailureProgressEventCrossVersionTest.groovy]: This was needed for the unused import only; removed.

**Comment by donat** [comment]
@bot-gradle test rfm

**Comment by donat** [comment]
@marcphilipp thank you for the review!

**Comment by donat** [comment]
@bot-gradle test and merge

**Pull request merged**